Move add new stats card option to top app bar

ravishanker opened the pull request:
This PR adds a new settings (⚙️) option to the top app bar on Stats screen as shown below.

Fixes #15877 

| Existing | New |
|---|---|
| ![device-2022-02-02-114441](https://user-images.githubusercontent.com/990349/152075739-3f46d36c-5461-4f91-ac50-3896bcd2af69.png) | ![device-2022-02-02-114330](https://user-images.githubusercontent.com/990349/152075774-3bbbc054-eae5-4fb1-9af4-2e236c8ef2e7.png) |


To test:

Test 1

- Launch app, then click on Stats either through quick links or Stats option below on My Site
- Scroll down if necessary, notice there's still + Add new stats card row (as in first image above)
- NOTE: you might need a site with already some stats cards in insights tab, otherwise you might see an empty stats view!  In that case please add some cards and check after that.

Test 2

- Launch app, then click on Stats either through quick links or Stats option below on My Site
- Notice a new settings icon on top right hand corner in the app bar (as in second image above)
- Click on the icon, it should launch stats management view, choose an option and hit save
- It should land back in insights tab and a new card selected in the previous step should be added (scroll down if necessary)
- Tap on any of the other tabs like Day, Week, Month or Year.  Notice the Settings icon is hidden when on those tabs.
- Return to Insights tab, settings icon should be visible again.


## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested different paths

3. What automated tests I added (or what prevented me from doing so)
Ran existing unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Move add new stats card to top app bar

- add new stats option menu
- handle menu in stats fragment
- handle navigation action in view model and fragment

ravishanker committed:
Remove existing add new stats card

Removed use case that was adding add new stats card

ravishanker committed:
add tracking for add new stats button

ravishanker committed:
Delete ManagementControlUseCase.kt

Remove unused class

ravishanker committed:
Supress detekt issues

Suppressing for now, in fact removed a param,  might look at refactoring opportunity later

ravishanker committed:
Update RELEASE-NOTES.txt

ravishanker committed:
Update StatsViewModel.kt

added back missing dismiss new card

renanferrari reviewed (commented):
Thanks for working on this! 🙇‍♂️ I tested the changes and can confirm both scenarios work as expected. I do have a couple of observations though:

1. I believe the menu option should be hidden when we navigate to the other tabs.
2. It would also be nice to have this logic inside the view model so we can test it.

Additionally – and this isn't directly related to this PR – while I was testing this I started to reconsider whether we should remove the bottom row or not. Started a discussion about this here: p1643834840352609-slack-C02UJ9192TC

Everything else looks good to me.

ravishanker committed:
hide settings icon on other tabs except insights

renanferrari reviewed (commented):
Thanks for the changes! I can confirm the app bar action is hidden when you start out in the Insights tab and navigate to another tab. However, it doesn't work well when we start out from a different tab and/or rotate the screen:

https://user-images.githubusercontent.com/830056/152377730-8500f0b2-ece2-4fce-8288-5776a917df26.mp4


> Good catch, thank you 🙇🏻‍♂️

In addition to that, we have decided to not remove the bottom row, so we should undo those changes and make sure to update tracks events as described here: p1643884316814009-slack-C02UJ9192TC

> ✅ Reverted to keep Add new statst card as before

renanferrari commented:
After looking into #15896, I had an idea that may help solve the issue I mentioned above. 

What do you think about moving `onCreateOptionsMenu` and `onOptionsItemSelected` to `StatsListFragment` and then check if the menu can be shown with `setHasOptionsMenu(statsSection == StatsSection.INSIGHTS)`?


>  I actually first put this into StatsListFragment, but it didn't work well, thought that may be due to it being inside a tab and tabs inside StatsFragment.   Now though, it worked like a charm.  Thank you.

ravishanker committed:
Move stats menu to StatsListFragment

Moving from StatsFragment to StatsListFragment for better handling of hiding on other tabs

ravishanker committed:
Remove stats settings menu from StatsFragment

Removing as it is now moved to StatsListFragment

ravishanker committed:
Revert "Delete ManagementControlUseCase.kt"

This reverts commit 28b67db38aa35bb26457933516aaab167b160abf.

ravishanker committed:
Revert "Remove existing add new stats card"

This reverts commit 28c4c71f453b6e9f01fe3e64bbb185329dce8ed9.

ravishanker committed:
Update StatsModule.kt

ravishanker committed:
Update ManagementControlUseCase.kt

add source param to tracking

ravishanker committed:
Merge branch 'trunk' into Move-add-new-stats-card-option-to-app-bar

renanferrari reviewed (approved):
Thanks for making these changes! I can confirm everything works as expected for me now 🙌

@ravishanker Not related to this PR specifically, but I noticed you replied to my previous comments by editing them. This is not ideal because I don't get notifications for any of your answers, and they are not immediately visible to me or anyone else who might need to check on this PR later on. In the future, it would be better to just leave a new comment instead and then quote the specific portions of the original comment you want to reply to.

LGTM! :shipit:
